### PR TITLE
Fix #337: Remove preprocess parameter from low-level functions

### LINKED ISSUE #337: Remove preprocess parameter from low-level functions
## Description

The `preprocess` parameter should be removed from low-level functions that are not `baselinenowcast()`. 

## Rationale

Low-level functions should focus on their core functionality and not include preprocessing logic. Users who want to preprocess data should do so explicitly before calling these functions, rather than having it as an optional parameter.

## Affected Functions

Functions that currently accept a `preprocess` parameter should be reviewed. The `baselinenowcast()` high-level wrapper can keep the parameter for convenience, but lower-level functions should expect users to handle preprocessing themselves if needed.

## Benefits

- Clearer separation of concerns
- More explicit data flow
- Reduced complexity in low-level functions
- Users have full control over preprocessing steps

### PR DESCRIPTION
**Pull request by seabbs**
## Summary

This PR simplifies the API by removing the `preprocess` parameter from low-level functions whilst keeping it available in high-level wrapper functions. This improves maintainability and clarity by ensuring preprocessing happens once at the entry point rather than being passed through the call chain.

## Changes

- **Removed `preprocess` parameter from low-level functions:**
  - `estimate_delay()`
  - `fill_triangle()`
  - `fill_triangles()`
  - `estimate_uncertainty_retro()`

- **Kept `preprocess` parameter in high-level wrapper functions:**
  - `baselinenowcast.reporting_triangle()`
  - `baselinenowcast.data.frame()`

- **Updated high-level wrappers** to apply preprocessing explicitly to the reporting triangle before calling low-level functions

- **Updated all tests** to reflect the new API, removing tests that specifically tested the `preprocess` parameter in low-level functions

- **Updated all documentation** including roxygen2 docstrings, examples, and generated man pages

- **Simplified examples** by removing preprocess-related use cases from low-level function examples

## Technical Details

The high-level functions (`baselinenowcast.reporting_triangle()` and `baselinenowcast.data.frame()`) now apply preprocessing to the reporting triangle before passing it to low-level estimation functions. This ensures:
- Preprocessing happens once at the entry point
- Low-level functions remain focused on their core responsibilities
- The call chain is simpler and easier to maintain
- Users still have full control over preprocessing via the high-level API

## Testing

All tests pass with the new API structure. Test results:
- 0 failures
- 0 warnings
- 3 skipped (expected edge cases)
- 2660 tests passing

## Related Issues

Fixes #337

## Summary by CodeRabbit

* **Refactor**
  * Reorganized preprocessing workflow: negative value preprocessing now applied at top-level nowcasting functions rather than intermediate estimation functions, simplifying the API.
  * Removed preprocessing parameter from several lower-level functions.

* **Documentation**
  * Updated examples and guidance to reflect the new preprocessing approach.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: R/baselinenowcast.R [hunk 1]
```diff
@@ -73,11 +73,17 @@ baselinenowcast <- function(data,
 #'   horizon 1 to the maximum horizon. Default is `NULL`, which will result in
 #'   computing the uncertainty parameters from the reporting triangle `data`.
 #'   See [estimate_uncertainty()] for more details.
+#' @param preprocess Function to apply to the reporting triangle before
+#'   estimation, or NULL to skip preprocessing. Default is
+#'   [preprocess_negative_values()], which handles negative values by
+#'   redistributing them to earlier delays. Set to NULL if you want to preserve
+#'   negative values. Custom preprocess functions must accept a `validate`
+#'   parameter (defaults to TRUE) to enable validation optimisation in internal
+#'   function chains.
```
[the remark this answers is not in the document]
Comment (kaitejohnson) [reply R/baselinenowcast.R]: Not sure if there are other places where we could sort of "flag" this is as a feature/option prior to making a vignette that demonstrates it.

File: R/baselinenowcast.R [hunk 2]
```diff
@@ -146,7 +157,7 @@ baselinenowcast.reporting_triangle <- function(
 
   if (is.null(uncertainty_params)) {
     uncertainty_params <- estimate_uncertainty_retro(
-      reporting_triangle = data,
+      reporting_triangle = processed_data,
       n_history_delay = tv$n_history_delay,
       n_retrospective_nowcasts = tv$n_retrospective_nowcasts,
       uncertainty_model = uncertainty_model
```
Comment (kaitejohnson) [inline R/baselinenowcast.R]: Same point below with the call to sample nowcasts, which is still using `data`. Shouldn't it be`processed_data`?

```r
nowcast_df <- sample_nowcasts(
    point_nowcast_matrix = pt_nowcast,
    reporting_triangle = processed_data,
    uncertainty_params = uncertainty_params,
    draws = draws,
    uncertainty_sampler = uncertainty_sampler,
    ...
  )
```

File: R/data.R [hunk 3]
```diff
@@ -114,13 +114,11 @@
...
 #' Use this example to understand:
 #' - How to work with negative corrections in delay distributions
-#' - The difference between `preprocess = NULL` (preserves negatives) and
-#'   `preprocess = preprocess_negative_values` (redistributes)
+#' - The impact of preprocessing negative values on delay estimation
 #' - How PMFs and CDFs behave with systematic downward corrections
 #'
 #' @seealso
 #' - [example_reporting_triangle] for a clean example without corrections
-#' - [estimate_delay()] with `preprocess` parameter
```
Comment (kaitejohnson) [inline R/data.R]: Should we add `[baselinenowcast.reporting_triangle()]` with `preprocess` parameter as this describes the `preprocess` parameter?

File: tests/testthat/test-estimate_delay.R [hunk 4]
```diff
@@ -182,38 +182,7 @@ test_that("estimate_delay handles diagonal reporting triangles", {
   )
 })
```
Comment (kaitejohnson) [inline tests/testthat/test-estimate_delay.R]: Tests were removed here but not added to `baselinenowcast.reporting_triangle()`. 
It would be good to test that in those functions when passing in a reporting triangle with negatives `baselinenowcast.reporting_triangle(preprocess = NULL)` preserves negative values, and with the default (`preprocess = `preprocess_negative_values()`) the function returns something with all positive entries. 

Also could add in there the test that was removed here for custom preprocessing?

File: R/data.R (hunk 3 above)
Comment (kaitejohnson) [reply R/data.R]: Have added

File: R/baselinenowcast.R (hunk 2 above)
Comment (kaitejohnson) [reply R/baselinenowcast.R]: fixed
